### PR TITLE
Implement a formatter to automatically fix common errors

### LINKED ISSUE #1: Format flag!
I love that! Could be possible implement "format" flag for repair automatically the files? that could be sweeeeeeeet 😄

### PR DESCRIPTION
**Pull request by MarshallOfSound**
This PR adds the `--auto-fix` option to the CLI interface that will automatically attempt to parse and fix JS standard syntax errors.

```bash
standard-markdown docs --auto-fix
```

It's currently working quite nicely and has a *dry run* method as well that will not overwrite files but write them to a temporary subdirectory `${cwd}_dry_run`.

Fixes #1 

/cc @zeke

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
**NOTE:** This is currently rebased on my `standard-exceptions` branch

**Comment by MarshallOfSound** [comment]
As a summary of changes from https://github.com/zeke/standard-markdown/pull/4/commits/1261fb5c43474f884e9dc45535627feee3e67296

* Modularized the formatting and linting components into logical files
* The formatter now runs the linter after formatting
* The linter now has the same progress feature the formatter has
* The linter now displays total errors as the final step before terminating
* The linters output is now a bit nicer and easier to read

TLDR: Adding the formatter made the code messy and a large single file so I modularized it

**Comment by MarshallOfSound** [comment]
Running this against electron/electron results in a drop from 1800 errors (in translations) to just over 200

**Comment by zeke** [comment]
Cool! A few thoughts:

Happy with the modularized code structure.

`standard` has replaced its `--format` option with `--fix` and now uses something different under the hood. See https://twitter.com/zeke/status/762137534683947008 -- I think it would be good to use a naming convention that's consistent with `standard`. Perhaps @feross or @flet can shed some more light on what has replaced `standard-format`.

`--safe-run` is a nice idea, but I don't think it's really necessary. I feel like most people using this tool are probably running it on a version-controlled codebase. If things go haywire, users can `git checkout .` to blow away changes.

**Comment by feross** [comment]
`standard --fix` uses ESLint's built-in formatting functionality. The nice thing about this vs. `standard-format` is that it doesn't require maintaining a whole separate config file for a different tool. It comes for free from the existing list of ESLint rules. It also appears to be a lot more reliable than `esformatter`, with perfect ES6 support.

One thing is that we probably should expose the `--fix` functionality via the `standard` Node.js API. I'll work on that.

`standard --fix` will be released with `standard` v8 on September 1, 2016. You can use it today by doing `npm install standard@beta --save-dev`.

**Comment by Flet** [comment]
...and @feross beats me again :) 

just adding "fix = true" to the eslint config if it comes in on the CLI should make it do its thing! ) 👍

**Comment by feross** [comment]
PR out here: https://github.com/Flet/standard-engine/pull/119/files

Will make it so you can just do:

```js
var standard = require('standard')
var files = ['a.js', 'b.js']
standard.lintFiles(files, {fix: true}, function (err, results) {
  console.log(results.errorCount + ' errors found')
})
```

**Comment by feross** [comment]
Come to think of it, you really need to be able to fix arbitrary source text. So you need `standard.lintText` to support `{fix: true}`. So here's a PR for that: https://github.com/Flet/standard-engine/pull/120

**Comment by feross** [comment]
Here's how you use it:

```js
var standard = require('standard')
standard.lintText('console.log(0);;;;', {fix: true}, function (err, result) {
  console.log(result.results[0].output) // logs 'console.log(0)'
})
```

**Comment by zeke** [comment]
Replaced by #5
